### PR TITLE
Feat/new network selector

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

File: src/lib/stores/networks.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  }
+
+  removeNetwork(chainId: string): void {
+    this.networks[this.userKey] = this.networks[this.userKey]?.filter(
+      (each) => each.chainId !== chainId
+    );
+  }
+
+  setPinnedNetworks(activeId: string, overId: string): void {
```
Comment (evilpeach) [inline src/lib/stores/networks.ts:65]: change this function name

**Review by songwongtp — APPROVED** [review]

File: src/lib/layout/network-menu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        event.preventDefault();
+        if (isOpen) {
+          onClose();
+        } else {
+          onOpen();
+        }
+      }
+    };
+    document.addEventListener("keydown", openSearchHandler);
+    return () => {
```
Comment (evilpeach) [inline src/lib/layout/network-menu/index.tsx:42]: remove explicit curly bracket

File: src/lib/layout/network-menu/NetworkMenuBody.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+import { NetworkAccodionPinned } from "./NetworkAccordionPinned";
+
+interface NetworkMenuBodyProps {
+  onClose: () => void;
+}
+
+const filterChains = (
+  chainIds: string[],
+  keyword: string,
+  type: "mainnet" | "testnet"
```
Comment (evilpeach) [inline src/lib/layout/network-menu/NetworkMenuBody.tsx:24]: ChainConfig["networkType"]

File: src/lib/layout/network-menu/NetworkMenuBody.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  const filteredPinnedNetworks = useMemo(() => {
+    if (!keyword) return [...pinnedNetworks];
+    return pinnedNetworks.filter(
+      (network) =>
+        CHAIN_CONFIGS[network.chainId]?.prettyName
+          .toLowerCase()
+          .includes(keyword.toLowerCase()) ||
+        network.chainId.toLowerCase().includes(keyword.toLowerCase())
+    );
+  }, [pinnedNetworks, keyword]);
```
Comment (evilpeach) [inline src/lib/layout/network-menu/NetworkMenuBody.tsx:74]: Can we utilize `filterChains` here? I think it's very similar by just making type argument to optional.

File: src/lib/layout/network-menu/network-card/NetworkIcons.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import { CustomIcon } from "lib/components/icon";
+import { useNetworkStore } from "lib/providers/store";
+
+interface NetworkIconsProps {
+  chainId: string;
+  isSelected: boolean;
+  isDraggable?: boolean;
+}
+
+export const NetworkIcons = observer(
```
Comment (evilpeach) [inline src/lib/layout/network-menu/network-card/NetworkIcons.tsx:16]: rename to NetworkCTAs

**Review by evilpeach — APPROVED** [review]

**Pull request merged**